dashboard/src/shared/Namespace.ts

```diff
@@ -52,11 +51,35 @@ export const definedNamespaces = {
   all: "_all",
 };
 
-export function getCurrentNamespace(currentNS: string, availableNS: string[]) {
+// The namespace information will contain a map[cluster]:namespace with the default namespaces
+const namespaceKey = "kubeapps_namespace";
+
+function getStoredNamespace(cluster: string) {
+  const ns = localStorage.getItem(namespaceKey) || "{}";
+  return JSON.parse(ns)[cluster] || "";
+}
+
+export function setStoredNamespace(cluster: string, namespace: string) {
+  const ns: string = localStorage.getItem(namespaceKey) || "{}";
+  const nsObject = JSON.parse(ns);
```

Comment:
Shouldn't we be handling exceptions (`SyntaxError`) both here and 5 lines up? (just defaulting to `{}` if there is a syntax error, which is perhaps what you intended?)

Reply:
In theory, this info is not something the user will modify on their own so a `SyntaxError` won't happen but it's true that if someone tamper this data, it may cause a DoS.